code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -552,31 +552,22 @@
 /datum/gas_reaction/bzformation/react(datum/gas_mixture/air)
 	var/list/cached_gases = air.gases
 	var/pressure = air.return_pressure()
-	// This slows down in relation to pressure, very quickly. Please don't expect it to be anything more then a snail
+	var/volume = air.return_volume()
+	var/pressure_effciency = 1/pressure
+	var/ratio_efficency = min(cached_gases[/datum/gas/nitrous_oxide][MOLES]/cached_gases[/datum/gas/plasma][MOLES],1)
+	var/bz_formed = min(pressure_effciency*ratio_efficency*volume/100, cached_gases[/datum/gas/nitrous_oxide][MOLES], cached_gases[/datum/gas/plasma][MOLES]*INVERSE(2))
 
-	// Bigger is better for these two values.
-	var/pressure_efficiency = (0.1 * ONE_ATMOSPHERE) / pressure // More pressure = more bad
-	var/ratio_efficiency = min(cached_gases[/datum/gas/nitrous_oxide][MOLES] / cached_gases[/datum/gas/plasma][MOLES], 1) // Malus to production if more plasma than n2o.
-
-	var/reaction_efficency = min(pressure_efficiency * ratio_efficiency, cached_gases[/datum/gas/nitrous_oxide][MOLES], cached_gases[/datum/gas/plasma][MOLES] * INVERSE(2))
-
-	if ((cached_gases[/datum/gas/nitrous_oxide][MOLES] - reaction_efficency < 0 )|| (cached_gases[/datum/gas/plasma][MOLES] - (2 * reaction_efficency) < 0) || reaction_efficency <= 0) //Shouldn't produce gas from nothing.
+	if ((cached_gases[/datum/gas/nitrous_oxide][MOLES] - bz_formed < 0 )|| (cached_gases[/datum/gas/plasma][MOLES] - (2 * bz_formed) < 0) || bz_formed <= 0)
```

Comment:
same here with the ||.
I don't think there's any reason to wrap everything in ()'s either
oh and was the comment still applicable?

Reply:
This was just copied from how it was before. I'll clean it up

Reply:
After fixing the INVERSEs in the bz_formed calculation I'm not sure this is really needed as the reaction should always consume an amount equal to or lower than the remaining plasma and n2o

Reply:
It's effectively a sanity check.
It would be worth investigating if it's one that even really needs to exist, but the pattern is used so widely across reaction code that it'd be better done separately